src/path.c

```diff
@@ -236,6 +236,154 @@ void twin_path_arc(twin_path_t *path,
     twin_path_set_matrix(path, save);
 }
 
+static twin_angle_t vector_angle(twin_fixed_t ux,
+                                 twin_fixed_t uy,
+                                 twin_fixed_t vx,
+                                 twin_fixed_t vy)
+{
+    twin_fixed_t dot = twin_fixed_mul(ux, vx) + twin_fixed_mul(uy, vy);
+
+    twin_fixed_t ua =
+        twin_fixed_sqrt(twin_fixed_mul(ux, ux) + twin_fixed_mul(uy, uy));
+    twin_fixed_t va =
+        twin_fixed_sqrt(twin_fixed_mul(vx, vx) + twin_fixed_mul(vy, vy));
+
+    /* cos(theta) = (u ⋅ v) / (|u| * |v|) */
+    twin_fixed_t cos_theta = twin_fixed_div(dot, twin_fixed_mul(ua, va));
+    twin_fixed_t cross = twin_fixed_mul(ux, vy) - twin_fixed_mul(uy, vx);
+    twin_angle_t angle = twin_acos(cos_theta);
+    return (cross < 0) ? -angle : angle;
+}
+
+typedef struct {
+    twin_fixed_t cx, cy;
+    twin_angle_t start, extent;
+} twin_ellipse_param_t;
+
+static twin_ellipse_param_t get_center_parameters(twin_fixed_t x1,
+                                                  twin_fixed_t y1,
+                                                  twin_fixed_t x2,
+                                                  twin_fixed_t y2,
+                                                  bool fa,
+                                                  bool fs,
+                                                  twin_fixed_t rx,
+                                                  twin_fixed_t ry,
+                                                  twin_fixed_t phi)
+{
+    twin_fixed_t sin_phi = twin_sin(phi);
+    twin_fixed_t cos_phi = twin_cos(phi);
+    /* Simplify through translation/rotation */
+    twin_fixed_t x =
+        twin_fixed_mul(cos_phi, twin_fixed_mul(x1 - x2, TWIN_FIXED_HALF)) +
+        twin_fixed_mul(sin_phi, twin_fixed_mul(y1 - y2, TWIN_FIXED_HALF));
+
+    twin_fixed_t y =
+        twin_fixed_mul(-sin_phi, twin_fixed_mul(x1 - x2, TWIN_FIXED_HALF)) +
+        twin_fixed_mul(cos_phi, twin_fixed_mul(y1 - y2, TWIN_FIXED_HALF));
+
+    twin_xfixed_t x_x = twin_fixed_to_xfixed(x);
+    twin_xfixed_t y_x = twin_fixed_to_xfixed(y);
+    twin_xfixed_t rx_x = twin_fixed_to_xfixed(rx);
+    twin_xfixed_t ry_x = twin_fixed_to_xfixed(ry);
+    twin_xfixed_t px_x = twin_xfixed_mul(x_x, x_x);
+    twin_xfixed_t py_x = twin_xfixed_mul(y_x, y_x);
+    twin_xfixed_t prx_x = twin_xfixed_mul(rx_x, rx_x);
+    twin_xfixed_t pry_x = twin_xfixed_mul(ry_x, ry_x);
+    twin_xfixed_t p_ry_divided_rx_x = twin_xfixed_div(pry_x, prx_x);
```

Comment:
`pry_divided_prx_x`: I'm unsure if there's a better way to name these variables.
The `p` in `p_ry_divided_rx_x` represents the `p` of `pry_x` and `prx_x`.
However, in `px_x`, the `p` means multiplication itself.

Reply:
`p_ry_divided_rx_x` is $(ry)^2/(rx)^2 = (ry/rx)^2$, `px_x` is $x^2$, the `p` represents power of 2 here.

Reply:
Ok!